### PR TITLE
Remove lots of minor dead code, refactor solr context

### PR DESCRIPTION
**Pull request by kanishka-azimi**
You can ignore all the minor dead code removal and removing of Has instances.

The solr context refactor is worth reviewing. I think this is universally better than what was before, so I don't expect any blocking objections. We might be able to indicate how we want to improve this further in later refactorings though. (Can probably factor out context into context needed for model layer and general context needed)

- [x] Run tests once locally with solr test enabled.

### CODE REVIEW

File: src/Context.hs [hunk 1]
```diff
@@ -7,6 +7,9 @@ module Context
   , BackgroundContext(..)
   , BackgroundContextM
   , withBackgroundContextM
```
Comment (kanishka-azimi) [inline src/Context.hs]: First part of changes here.

File: src/Service/Periodic.hs [hunk 2]
```diff
@@ -41,7 +41,7 @@ run p = runContextM $ withReaderT BackgroundContext $ do
   focusIO $ logMsg t ("periodic running: " ++ show p)
   cleanTokens
   updateVolumeIndex
-  updateIndex
+  withReaderT (mkSolrIndexingContext . backgroundContext) updateIndex
```
Comment (kanishka-azimi) [inline src/Service/Periodic.hs]: Second part of changes

File: src/Solr/Index.hs [hunk 3]
```diff
@@ -172,7 +172,7 @@ writeDocuments [] = return ()
 writeDocuments d =
   writeBlock $ BSL.toStrict $ BSB.toLazyByteString $ foldMap (("},\"add\":{\"doc\":" <>) . JSON.fromEncoding . JSON.value . JSON.toJSON) d
 
-writeUpdate :: ReaderT BackgroundContext (InvertM BS.ByteString) () -> ReaderT BackgroundContext (InvertM BS.ByteString) ()
+writeUpdate :: ReaderT SolrIndexingContext (InvertM BS.ByteString) () -> ReaderT SolrIndexingContext (InvertM BS.ByteString) ()
```
Comment (kanishka-azimi) [inline src/Solr/Index.hs]: Third part of changes

File: test/ModelTest.hs [hunk 4]
```diff
@@ -426,7 +426,7 @@ test_15 = Test.stepsWithTransaction "test_15" $ \step cn2 -> do
 ------- search -------------
 test_16 :: TestTree
 test_16 = ignoreTest $ -- TODO: enable this inside of nix build with solr binaries and core installed in precheck
-    Test.stepsWithResourceAndTransaction "test_16" $ \step ist cn2 -> do
+    Test.stepsWithTransaction "test_16" $ \step cn2 -> do
```
Comment (kanishka-azimi) [inline test/ModelTest.hs]: Fourth part of changes

File: test/TestHarness.hs [hunk 5]
```diff
@@ -1,6 +1,7 @@
 module TestHarness
     (
-      ContextFor(..)
+      mkSolrIndexingContextSimple
```
Comment (kanishka-azimi) [inline test/TestHarness.hs]: Fifth part of changes

File: src/Model/AssetRevision/Types.hs [hunk 6]
```diff
@@ -10,5 +9,5 @@ data AssetRevision = AssetRevision
   , revisionOrig :: !Asset
   }
 
-makeAssetRevision :: Asset -> Asset -> AssetRevision
-makeAssetRevision o a = AssetRevision a o
+-- makeAssetRevision :: Asset -> Asset -> AssetRevision
+-- makeAssetRevision o a = AssetRevision a o
```
Comment (chreekat) [inline src/Model/AssetRevision/Types.hs]: .

File: src/Model/AssetSegment/Types.hs [hunk 7]
```diff
@@ -147,8 +146,8 @@ instance Has (Id Asset) Excerpt where
   view = view . excerptAsset
 instance Has Volume Excerpt where
   view = view . excerptAsset
-instance Has (Id Volume) Excerpt where
-  view = view . excerptAsset
+{- instance Has (Id Volume) Excerpt where
+  view = view . excerptAsset -}
```
Comment (chreekat) [inline src/Model/AssetSegment/Types.hs]: .

File: src/Model/Format/Types.hs [hunk 8]
```diff
@@ -7,12 +7,11 @@ module Model.Format.Types
 import qualified Data.ByteString as BS
 import Data.Function (on)
 import Data.Int (Int16)
-import Data.Ord (comparing)
 import qualified Data.Text as T
 import Instances.TH.Lift ()
 import Language.Haskell.TH.Lift (deriveLift)
 
-import Model.Kind
+-- import Model.Kind
```
Comment (chreekat) [inline src/Model/Format/Types.hs]: .

File: src/Model/Format/Types.hs [hunk 9]
```diff
@@ -23,17 +22,12 @@ data Format = Format
   , formatExtension :: [BS.ByteString] -- TODO: nonempty list
   , formatName :: T.Text
   }
-  -- deriving (Show)
 
-instance Kinded Format where
-  kindOf _ = "format"
+-- instance Kinded Format where
+--  kindOf _ = "format"
```
Comment (chreekat) [inline src/Model/Format/Types.hs]: .

File: src/Model/Format/Types.hs [hunk 10]
```diff
@@ -23,17 +22,12 @@ data Format = Format
...
 
 instance Eq Format where
   (==) = on (==) formatId
-  (/=) = on (/=) formatId
```
Comment (chreekat) [inline src/Model/Format/Types.hs]: ... nice

File: src/Model/Record/Types.hs [hunk 11]
```diff
@@ -73,46 +72,30 @@ data ParticipantRecord =
...
 data Measure = Measure
   { measureRecord :: Record
   , measureMetric :: Metric
   , measureDatum :: !MeasureDatum
   }
 
-instance Kinded Measure where
-  kindOf _ = "measure"
+-- instance Kinded Measure where
+--   kindOf _ = "measure"
```
Comment (chreekat) [inline src/Model/Record/Types.hs]: .

File: src/Model/RecordSlot/Types.hs [hunk 12]
```diff
@@ -45,8 +45,8 @@ instance Has (Id Category) RecordSlot where
   view = view . slotRecord
 instance Has Volume RecordSlot where
   view = view . slotRecord
-instance Has (Id Volume) RecordSlot where
-  view = view . slotRecord
+{- instance Has (Id Volume) RecordSlot where
+  view = view . slotRecord -}
```
Comment (chreekat) [inline src/Model/RecordSlot/Types.hs]: .

File: src/Model/Slot/Types.hs [hunk 13]
```diff
@@ -46,8 +46,8 @@ instance Kinded Slot where
 
 instance Has Container Slot where
   view = slotContainer
-instance Has (Id Model.Volume.Types.Volume) Slot where
-  view = (view . slotContainer)
+{- instance Has (Id Model.Volume.Types.Volume) Slot where
+  view = (view . slotContainer) -}
```
Comment (chreekat) [inline src/Model/Slot/Types.hs]: .

File: src/Model/Volume/Types.hs [hunk 14]
```diff
@@ -41,15 +40,12 @@ data Volume = Volume
...
 
 instance Kinded Volume where
   kindOf _ = "volume"
 
-instance Has (Id Volume) VolumeRow where
-  view = volumeId
-instance Has (Id Volume) Volume where
-  view = (view . volumeRow)
+{- instance Has (Id Volume) Volume where
+  view = (volumeId . volumeRow) -}
```
Comment (chreekat) [inline src/Model/Volume/Types.hs]: .

**Pull request merged**